pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/VariableCanBeInlined.xml

```diff
@@ -0,0 +1,1010 @@
+    </test-code>
+    <test-code>
+        <description>Throw inside for loop block</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>4</expected-linenumbers>
+        <code><![CDATA[
+public class Test {
+    void test() {
+        for (int i = 0; i < 1; i++) {
+            RuntimeException e = new RuntimeException();
+            throw e;
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Throw using lambda expression - no violation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.util.function.Supplier;
+
+public class Test {
+    void test() {
+        Supplier<RuntimeException> s = () -> new RuntimeException();
+        throw s.get();
+    }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Throw in ternary - no violation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Test {
+    void test(boolean condition) {
+        throw condition ? new IllegalArgumentException() : new RuntimeException();
+    }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Nested block with unnecessary throw variable</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>4</expected-linenumbers>
+        <code><![CDATA[
+public class Test {
+    void test() {
+        {
+            RuntimeException e = new RuntimeException();
+            throw e;
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Inner class with throw - no violation</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>4</expected-linenumbers>
+        <code><![CDATA[
+public class Test {
+    class Inner {
+        void test() {
+            RuntimeException e = new RuntimeException();
+            throw e;
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Exception assigned and rethrown inside catch - no violation</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>6</expected-linenumbers>
+        <code><![CDATA[
+public class Test {
+    void test() {
+        try {
+            something();
+        } catch (Exception ex) {
+            Exception e = ex;
+            throw e;
+        }
+    }
+    void something() throws Exception {}
+}
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Multiple throw variables but one is unnecessary</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>4</expected-linenumbers>
+        <code><![CDATA[
+public class Test {
+    void test() {
+        RuntimeException e = new RuntimeException();
+        RuntimeException f = new RuntimeException();
+        throw f;
+    }
+}
+        ]]></code>
+    </test-code>
```

Comment:
This looks duplicated - the same test case is in line 460 "Multiple variables but only one used in throw". Please remove this test case.